eduvmstorebackend/eduvmstore/db/operations/roles.py

```diff
@@ -0,0 +1,36 @@
+import uuid
+from django.db import transaction
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from eduvmstore.db.models import Role
+
+def create_role(data: dict) -> Role:
```

Comment:
dict == hashmap? Ist das normall in python eine Hashmap dem konstruktor zu geben?
Ich hätte jetzt einfach die ganzen Atribute übergeben.

Reply:
"Statt direkt jedes Attribut manuell an den Konstruktor zu übergeben (wie es in der Anmerkung vorgeschlagen wurde), ermöglicht die Verwendung eines Dictionaries, eine kompakte und flexible Methode zu haben, um Daten zu übergeben. Es ist jedoch auch absolut möglich, die Werte direkt als separate Argumente zu übergeben, was der traditionelle Ansatz wäre. Der Vorteil eines Dictionaries ist jedoch, dass man dynamischer und flexibler mit den Daten arbeiten kann (z. B. wenn optionale Felder vorhanden sind)."
Ich würde es daher so lassen, oder?

Reply:
Ok